Src/AutoFixture.NUnit3.UnitTest/TestNameStrategiesFixture.cs

```diff
@@ -0,0 +1,85 @@
+﻿using NUnit.Framework.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Ploeh.AutoFixture.NUnit3.UnitTest
+{
+    public class TestNameStrategiesFixture
+    {
+        #region Custom attribute definitions
+        public const int InjectedIntValue = 1;
+        public const string InjectedStringValue = "foo";
+
+        private static IFixture CreateFixtureWithInjectedValues()
+        {
+            var result = new Fixture();
+            // Make so that fixed values will be returned
+            result.Inject(InjectedIntValue);
+            result.Inject(InjectedStringValue);
+            return result;
+        }
+
+        public class AutoDataFixedNameAttribute : AutoDataAttribute
+        {
+            public AutoDataFixedNameAttribute()
+            {
+                TestMethodBuilder = new FixedNameTestMethodBuilder();
+            }
+        }
+
+        public class AutoDataDefaultNameAttribute : AutoDataAttribute
+        {
+            public AutoDataDefaultNameAttribute() : base(CreateFixtureWithInjectedValues())
```

Comment:
> The purpose of this test was to test what it the default strategy for non-regression testing - eg. make sure that the name won't be altered after people update their NuGet package.

Currently you have two kind of attributes - `AutoDataFixedNameAttribute` and `AutoDataDefaultNameAttribute`. In this scenario context it's better to have `AutoDataFixedNameAttribute` and `AutoDataVolatileNameAttribute` - to cover all the cases individually.

If you wish add tests for regression, we should:
- Assert the default strategy - already done (`AutoDataAttributeUsesRealValueByDefault()` and `InlineAutoDataAttributeUsesRealValueByDefault()`).
- Add yet another set of tests where you test the `AutoDataAttribute` directly and assert that name is mutating by default.

I'd say that the last point is not obligatory, but if you wish - feel free to add it.

P.S. Please add reply comments in place, rather than in global scope - it's easier to track them 😉

Reply:
The main issue with current approach is that when we change the default strategy, all those tests will become failing and no tests will cover the `VolatileTestBuilder` anymore.